atlp-rwanda/e-commerce-bitcrafters-bn
# #[187364724] - A successfully registered user should receive an email to verify their account email

**What does this PR do**
Users need to verify their email to avoid spams

**Description of Task to be completed?**
- Added nodemailer to help in sending emails
- Created a reusable component for sending emails
- Updated signup and login controllers to handle email verification on signup and check for verification status on login
- Added email verifying component to handle email verification logic
- Added endpoints for email verification `/user/verify/:token`

 **How should this be manually tested?**
- Fire up your application or use the hosted server
- You can use use swagger docs to test it by going to `/users/signup`
- Add your real email address and all necessary information for signup
- If successful, check your gmail or any email platform you use (for the email you used during signup)
- Click on the verification link provided
- Test the verification on swagger on `/users/verify/:token`

![Screenshot 2024-04-30 113339](https://github.com/atlp-rwanda/e-commerce-bitcrafters-bn/assets/39400022/cf32f96f-dfef-4581-88ad-bfc061949b52)
![Screenshot 2024-04-30 113406](https://github.com/atlp-rwanda/e-commerce-bitcrafters-bn/assets/39400022/25d21201-e8c1-4637-ba9c-27d63912b286)
![Screenshot 2024-04-30 113452](https://github.com/atlp-rwanda/e-commerce-bitcrafters-bn/assets/39400022/8d5b24f6-d2cf-4661-99a7-0bf824d767c4)
![Screenshot 2024-04-30 113720](https://github.com/atlp-rwanda/e-commerce-bitcrafters-bn/assets/39400022/3da72b3c-be97-4c78-9893-c66197433bfb)
![Screenshot 2024-04-30 113746](https://github.com/atlp-rwanda/e-commerce-bitcrafters-bn/assets/39400022/c49e1cea-56da-4e16-9f33-e936c2226d05)







 **What are the relevant pivotal tracker/Trello stories?**
#187364724
